analyzers/tests/SonarAnalyzer.UnitTest/Rules/UseLambdaParameterInConcurrentDictionaryTest.cs

```diff
@@ -27,9 +27,10 @@ namespace SonarAnalyzer.UnitTest.Rules;
 public class UseLambdaParameterInConcurrentDictionaryTest
 {
     [TestMethod]
-    public void UseLambdaParameterInConcurrentDictionary_CS() =>
-        new VerifierBuilder<CS.UseLambdaParameterInConcurrentDictionary>().AddPaths("UseLambdaParameterInConcurrentDictionary.cs")
+    public void UseLambdaParameterInConcurrentDictionary_CSharp8() =>
+        new VerifierBuilder<CS.UseLambdaParameterInConcurrentDictionary>().AddPaths("UseLambdaParameterInConcurrentDictionary.CSharp8.cs")
             .AddReferences(MetadataReferenceFacade.SystemCollections)
+            .WithOptions(ParseOptionsHelper.FromCSharp8)
```

Comment:
We are testing for CSharp8 onwards since:
- lambda expressions were introduced starting from CSharp6.
- CSharp6 and CSharp7 are producing a compile error CS0136 on the lambda expressions that declare a parameter with the same name as an existing local variable (they will nevertheless not raise any S6612 with the new logic)